### PR TITLE
Replace emoji-mart with emoji-picker-element

### PR DESCRIPTION
**Pull request by ErikMichelson**
### Component/Part
Editor -> Toolbar -> Emoji picker

### Description
This PR replaces emoji-mart with emoji-picker-element. Please note, that this doesn't already include the ForkAwesome icons correctly as we wait for response from the emoji-picker-element developers.

### Steps

- [x] added implementation
- [x] added / updated tests
- [ ] added / updated documentation
- [x] extended changelog

### Related Issue(s)
#430 (not closing)

### CODE REVIEW

File: src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx [hunk 1]
```diff
@@ -1,47 +1,81 @@
...
+  useEffect(() => {
+    if (!pickerContainerRef.current || firstOpened.current) {
+      return
+    }
+    const picker = new Picker({
+      customEmoji: customEmojis,
+      dataSource: '/static/js/emoji-data.json'
+    })
+    const container = pickerContainerRef.current
+    picker.addEventListener('emoji-click', emojiClickListener)
```
Comment (DerMolly) [inline src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx:52]: remove this listener onUnload?

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Comment by DerMolly** [comment]
Also this PR should probably remove `externalTypes/emoji-mart`

**Comment by ErikMichelson** [comment]
It does remove the external types: https://github.com/codimd/react-client/pull/620/commits/b109987f4a536f27e7f638c160772b4a72179ac9#diff-134d226e34bcf57915f0e22064d5d18c

File: src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx (hunk 1 above)
Comment (ErikMichelson) [reply src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx:52]: This doesn't work unfortunately. Emojis get unselectable from the picker when applying the following change:

```diff
index 3b98108..d42c743 100644
--- a/src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx
+++ b/src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx
@@ -55,6 +55,9 @@ export const EmojiPicker: React.FC<EmojiPickerProps> = ({ show, onEmojiSelected,
     }
     container.appendChild(picker)
     firstOpened.current = true
+    return () => {
+      picker.removeEventListener('emoji-click', emojiClickListener)
+    }
   }, [pickerContainerRef, emojiClickListener, darkModeEnabled, twemojiStyle])

   useEffect(() => {
```

Comment (ErikMichelson) [reply src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx:52]: Apparently the useEffect expires already when the picker is opened.

Comment (DerMolly) [reply src/components/editor/editor-pane/tool-bar/emoji-picker/emoji-picker.tsx:52]: okay then I'll close this

**Review by DerMolly — APPROVED** [review]

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]
Something seems to be wrong.

image

**Comment by mrdrogdrog** [comment]
Interessting. It was my local setup that caused the bug. Partly.

I had a local installation of Twemoji to render emojis in my WM with the Name "Twemoji Mozilla". Chrome also uses this file, but can only use the black/white fallback. I really don't know why, because if you load a colored font via css then everything is okay.
The emoji picker has an built-in check to determine if the browser supports colored emojis. It renders an emoji onto two canvas in white and black. If the two renderings are the same (because the browser ignores the font-color and uses the color layers from the font), then the picker assumes, that the browser supports colored fonts.

Now comes the twisted part:
Our CSS imports the font file under the name "twemoji". This is my I see colored emojis. But the picker ignores the custom font-family and [uses a constant font-family for the test](https://github.com/nolanlawson/emoji-picker-element/blob/master/src/picker/utils/testColorEmojiSupported.js). And in this font-family const was?.. right. "Twemoji Mozilla".
So my browser used the local font to do the test, but uses our provided font to do the rendering.

**Comment by mrdrogdrog** [comment]
https://github.com/codimd/react-client/pull/664 fixes this issue.

**Review by mrdrogdrog — APPROVED** [review]
I tested it in firefox and it looks good. It's your choice if this PR gets merged before or after #664

**Pull request merged**